### ISSUE TITLE
AbstractMethodError - Using MoPub custom event network.

### ISSUE THREAD

**Issue by InterestingTime** [mobfox/MobFox-Android-SDK#49]
When using a custom event network on MoPub for MobFox I get this error:

```
                                                           java.lang.AbstractMethodError: abstract method "void com.mopub.nativeads.CustomEventNative.loadNativeAd(android.content.Context, com.mopub.nativeads.CustomEventNative$CustomEventNativeListener, java.util.Map, java.util.Map)"
                                                               at com.mopub.nativeads.CustomEventNativeAdapter.loadNativeAd(CustomEventNativeAdapter.java:47)
                                                               at com.mopub.nativeads.MoPubNative.onAdLoad(MoPubNative.java:232)
                                                               at com.mopub.nativeads.MoPubNative.access$000(MoPubNative.java:37)
                                                               at com.mopub.nativeads.MoPubNative$2.onSuccess(MoPubNative.java:93)
                                                               at com.mopub.network.AdRequest.deliverResponse(AdRequest.java:310)
                                                               at com.mopub.network.AdRequest.deliverResponse(AdRequest.java:46)
                                                               at com.mopub.volley.ExecutorDelivery$ResponseDeliveryRunnable.run(ExecutorDelivery.java:99)
                                                               at android.os.Handler.handleCallback(Handler.java:739)
                                                               at android.os.Handler.dispatchMessage(Handler.java:95)
                                                               at android.os.Looper.loop(Looper.java:148)
                                                               at android.app.ActivityThread.main(ActivityThread.java:5461)
                                                               at java.lang.reflect.Method.invoke(Native Method)
                                                               at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                               at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

Using com.mopub:mopub-sdk-native-static:4.11.0 and MobFox SDK 3.1.2 with the latest MobFox-Android-MoPub-Adapter.jar

**Comment by InterestingTime** [comment]
In file MoPubNativeAdapter.java.

Caused by:

`protected void loadNativeAd(@NonNull Activity activity, @NonNull CustomEventNativeListener customEventNativeListener, @NonNull Map<String, Object> localExtras, @NonNull Map<String, String> serverExtras) {`

in MoPub 4.11 it uses Context, not Activity:

`protected void loadNativeAd(@NonNull Context context, @NonNull CustomEventNativeListener customEventNativeListener, @NonNull Map<String, Object> localExtras, @NonNull Map<String, String> serverExtras) {`

**Comment by nabriski** [comment]
Hi,
We will upload a fix on Sunday. As I know this code was tested, I wonder how this bug has crept in.

**Comment by InterestingTime** [comment]
If possible it would also be good to rename the file slightly, only reason being, I have already included the current version in my latest APK and I had to disable MobFox network in MoPub to stop the crashes. When I upload a new APK and re-enable the network in MoPub, anybody on my older APK will continue to get crashes as it will reference the old MoPubNativeAdapter.java file still. If not I may be able to target only my newer APK.

**Comment by nabriski** [comment]
Sure, we will rename it.

**Comment by InterestingTime** [comment]
Great! Thank you

**Comment by asafgaber** [comment]
Hi,
fixed the issue. the fixed adapter is in latest release 3.1.5.
fixed adapter native class 'MoPubNativeAdAdapter' - dont forget to change in mopub custom network.

**Comment by InterestingTime** [comment]
Thanks, I can't find the latest MobFox-Android-MoPub-Adapter.jar though? In https://github.com/mobfox/MobFox-Android-SDK/commit/e518d9252ad5737c140ac8807f622c05bec4f1c6 it shows the adMob adapter being changed but not the MoPub?

**Comment by InterestingTime** [comment]
Yeah, looks like you might of uploaded the AdMob adapter by accident?

**Comment by asafgaber** [comment]
youre right. my bad. the fix in 3.1.6

**Comment by InterestingTime** [comment]
no problem :) thanks
